### PR TITLE
fix(useClipboard): use legacy way when without permission

### LINKED ISSUE #2494: useClipboard doesn't fallback to legacy in iframe
### Describe the bug

I received this error:
> Uncaught (in promise) DOMException: The Clipboard API has been blocked because of a permissions policy applied to the current document. See https://goo.gl/EuHzyv for more details.

useClipboard checks to see if clipboard is supported, but doesn't check to see if permissions are set to allow it.

https://web.dev/async-clipboard/#permissions-policy-integration

```
const queryOpts = { name: 'clipboard-write', allowWithoutGesture: false };
const permissionStatus = await navigator.permissions.query(queryOpts);
```

In my case, I cannot change the permissions on the iframe, so I have to use legacy.  To workaround this issue, I am forcing the legacy by doing:
```
// set navigator to null or false to force legacy
const { copy } = useClipboard({ source: sourceForClipboardCopy, legacy: true, navigator: <any>false }); 
```

### Reproduction

https://stackblitz.com/edit/vitejs-vite-jcxqdg

### System Info

```Shell
"@vueuse/core": "^9.6.0",
"vue": "^3.2.25"
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vueuse/vueuse/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vueuse.org/guide).
- [X] Check that there isn't [already an issue](https://github.com/vueuse/vueuse/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a VueUse issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to https://github.com/vuejs/core instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vueuse/vueuse/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### PR DESCRIPTION
**Pull request by Cat1007**
fixes #2494

### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

<details>
<summary><strong>⚠️ Slowing down new functions</strong></summary>
<br>

> **Warning**: **Slowing down new functions**
>
> As the VueUse audience continues to grow, we have been inundated with an overwhelming number of feature requests and pull requests. As a result, maintaining the project has become increasingly challenging and has stretched our capacity to its limits. As such, in the near future, we may need to slow down our acceptance of new features and prioritize the stability and quality of existing functions. **Please note that new features for VueUse may not be accepted at this time.** If you have any new ideas, we suggest that you first incorporate them into your own codebase, iterate on them to suit your needs, and assess their generalizability. If you strongly believe that your ideas are beneficial to the community, you may submit a pull request along with your use cases, and we would be happy to review and discuss them. Thank you for your understanding.

</details>

---

### Description

use legacy methods as a fallback when clipboard is supported but without permission

### Additional context

---

### <samp>🤖 Generated by Copilot at f04e403</samp>

This pull request enhances the `useClipboard` function with permission checks. It uses the `usePermission` function to query and request the clipboard permission before reading or writing to the clipboard.

### <samp>🤖 Generated by Copilot at f04e403</samp>

* Import `usePermission` function to check clipboard access ([link](https://github.com/vueuse/vueuse/pull/3379/files?diff=unified&w=0#diff-104a20ea8a35b01c36721b36a81d3138312e27a997280d66290ce7e8a9bc466aR11))
* Declare `permissionRead` and `permissionWrite` variables to store the result of `usePermission` ([link](https://github.com/vueuse/vueuse/pull/3379/files?diff=unified&w=0#diff-104a20ea8a35b01c36721b36a81d3138312e27a997280d66290ce7e8a9bc466aR66-R67))
* Modify `updateText` function to conditionally call `navigator.clipboard.readText` based on `permissionRead` ([link](https://github.com/vueuse/vueuse/pull/3379/files?diff=unified&w=0#diff-104a20ea8a35b01c36721b36a81d3138312e27a997280d66290ce7e8a9bc466aL71-R74))
* Modify `copy` function to conditionally call `navigator.clipboard.writeText` based on `permissionWrite` ([link](https://github.com/vueuse/vueuse/pull/3379/files?diff=unified&w=0#diff-104a20ea8a35b01c36721b36a81d3138312e27a997280d66290ce7e8a9bc466aL86-R89))

### CODE REVIEW

**Comment by Cat1007** [comment]
@antfu  could you please approve the check

File: packages/core/useClipboard/index.ts [hunk 1]
```diff
@@ -62,13 +63,15 @@ export function useClipboard(options: UseClipboardOptions<MaybeRefOrGetter<strin
...
+  const permissionRead = usePermission('clipboard-read')
+  const permissionWrite = usePermission('clipboard-write')
   const isSupported = computed(() => isClipboardApiSupported.value || legacy)
   const text = ref('')
   const copied = ref(false)
   const timeout = useTimeoutFn(() => copied.value = false, copiedDuring)
 
   function updateText() {
-    if (isClipboardApiSupported.value) {
+    if (isClipboardApiSupported.value && permissionRead) {
```
Comment (posva) [inline packages/core/useClipboard/index.ts:74]: Should probably be

```suggestion
    if (isClipboardApiSupported.value && permissionRead.value !== 'denied') {
```

File: packages/core/useClipboard/index.ts [hunk 2]
```diff
@@ -83,7 +86,7 @@ export function useClipboard(options: UseClipboardOptions<MaybeRefOrGetter<strin
 
   async function copy(value = toValue(source)) {
     if (isSupported.value && value != null) {
-      if (isClipboardApiSupported.value)
+      if (isClipboardApiSupported.value && permissionWrite)
```
Comment (posva) [inline packages/core/useClipboard/index.ts:89]: Same as above:

```suggestion
      if (isClipboardApiSupported.value && permissionWrite.value !== 'denied')
```

**Review by posva — CHANGES_REQUESTED** [review]
You should probably test it goes for the legacy method when the permission is denied

**Comment by Cat1007** [comment]
> You should probably test it goes for the legacy method when the permission is denied

My mistake, thanks for your correction

**Review by antfu — APPROVED** [review]

**Comment by Mister-Hope** [comment]
This pr has a few bugs and I will open a new pr https://github.com/vueuse/vueuse/pull/3812 to fix this.

**Pull request merged**